index.js

```diff
@@ -45,8 +46,18 @@ module.exports = function(socket, opts) {
 
   function init() {
     socket.on('connect', function() {
-      queuedMessages.splice(0).forEach(bufferMessage);
-      signaller('connected');
+      if (!reconnected){
+        queuedMessages.splice(0).forEach(bufferMessage);
+        signaller('connected');
+      }
+      else {
+        console.log('pass as it is reconnect');
```

Comment:
I think I'd remove the `else` statement here and the `console.log` as I don't think it's going to give you much value beyond debugging the initial implementation.

Reply:
Agree. It's only when debugging purpose.